src/md/inc/VerifyLayouts.inc

```diff
@@ -273,23 +265,28 @@ BEGIN_TYPE(CGuidPoolHash, sizeof(CChainedHash<GUIDHASH>))
 FIELD(CGuidPoolHash, m_Pool, sizeof(void*))
 END_TYPE(CGuidPoolHash, sizeof(void*))
 
-
-BEGIN_TYPE_ESCAPED(MetaData::HotHeap, MetaData__HotHeap, 0)
-FIELD_ESCAPED(MetaData::HotHeap, MetaData__HotHeap, m_pHotHeapHeader, sizeof(void*))
-END_TYPE_ESCAPED(MetaData::HotHeap, MetaData__HotHeap, sizeof(void*))
+USING_ALIAS(MetaData__HotHeap, MetaData::HotHeap)
+BEGIN_TYPE(MetaData__HotHeap, 0)
+FIELD(MetaData__HotHeap, m_pHotHeapHeader, sizeof(void*))
+END_TYPE(MetaData__HotHeap, sizeof(void*))
 
 
 BEGIN_TYPE(StgPoolReadOnly, sizeof(StgPoolSeg) + sizeof(void*))                   //vtable pointer
 ALIGN_FIELD(StgPoolReadOnly, m_HotHeap, sizeof(MetaData::HotHeap), sizeof(void*))
 END_TYPE(StgPoolReadOnly, sizeof(void*))
 
-BEGIN_TYPE_ESCAPED(IGNORE_COMMAS(MapSHash<ULONG, ULONG>), MapSHash__ULONG__ULONG, sizeof(void*))
-FIELD_ESCAPED(IGNORE_COMMAS(MapSHash<ULONG, ULONG>), MapSHash__ULONG__ULONG, m_table, sizeof(void*))
-FIELD_ESCAPED(IGNORE_COMMAS(MapSHash<ULONG, ULONG>), MapSHash__ULONG__ULONG, m_tableSize, 4)
-FIELD_ESCAPED(IGNORE_COMMAS(MapSHash<ULONG, ULONG>), MapSHash__ULONG__ULONG, m_tableCount, 4)
-FIELD_ESCAPED(IGNORE_COMMAS(MapSHash<ULONG, ULONG>), MapSHash__ULONG__ULONG, m_tableOccupied, 4)
-FIELD_ESCAPED(IGNORE_COMMAS(MapSHash<ULONG, ULONG>), MapSHash__ULONG__ULONG, m_tableMax, 4)
-END_TYPE_ESCAPED(IGNORE_COMMAS(MapSHash<ULONG, ULONG>), MapSHash__ULONG__ULONG, sizeof(void*))
+USING_ALIAS(MapSHash__ULONG__ULONG, MapSHash<ULONG, ULONG>) // Create a using alias to avoid commas in the type name
+#ifdef _MSC_VER // Corresponds to `#ifdef _MSC_VER` in Target_MapSHash::ReadFrom(DataTargetReader & reader)
```

Comment:
There is no _MSC_VER in this Target_MapSHash::ReadFrom that I am aware of? https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/coreclr/src/md/datasource/targettypes.cpp#L133
Is there a missing file?

For cross-plat scenarios we also wouldn't be able to rely on having the same macros defined in host and target. There 
is a crude mechanism in place, [DataTargetReader.IsDefined()](https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/coreclr/src/md/datasource/targettypes.cpp#L440) that was intended to handle conditional layouts based on defines present in the target. The current implementation relies on the target to [declare any defines that are relevant](https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/coreclr/src/debug/ee/debugger.h#L2931), but you could also imagine infering additional ones using a heuristic.

Reply:
> There is no _MSC_VER in this Target_MapSHash::ReadFrom that I am aware of? https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/coreclr/src/md/datasource/targettypes.cpp#L133
> Is there a missing file?

Release 5.0 doesn't have it, Steve removed it in dotnet/runtime#26980. But given this is 3.1:
https://github.com/dotnet/coreclr/blob/e8b17841cb5ce923aec48a1b0c12042d445d508f/src/md/datasource/targettypes.cpp#L138

Reply:
Aha, thanks : ) In that case I think just the second part of the comment applies, this layout reading code in DBI executed on a Windows host will incorrectly assume Linux targets use the Windows layout rules.